### PR TITLE
Release a new version for Z-checker and fix a typo in SZ

### PR DESCRIPTION
**Pull request by disheng222**
Z-checker 0.6.0 is released

### CODE REVIEW

File: var/spack/repos/builtin/packages/z-checker/package.py [hunk 1]
```diff
@@ -12,10 +12,18 @@ class ZChecker(AutotoolsPackage):
...
+    url      = "https://github.com/CODARcode/Z-checker/releases/download/0.6.0/Z-checker-0.6.0.tar.gz"
+    git      = "https://github.com/CODARcode/Z-checker"
     maintainers = ['disheng222']
 
+
+    def url_for_version(self, version):
+        """provide url to ensure that download counting via github releases
+        works accurately"""
+        url = "https://github.com/CODARcode/Z-checker/releases/download/{0}/Z-checker-{0}.tar.gz"
+        return url.format(version)
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/z-checker/package.py]: Can you move the function below, under all the directives?

**Review by alalazo — CHANGES_REQUESTED** [review]
LGTM, just one minor request

File: var/spack/repos/builtin/packages/z-checker/package.py (hunk 1 above)
Comment (disheng222) [reply var/spack/repos/builtin/packages/z-checker/package.py]: Hi alalazo, 
I just committed a new version based on your suggestion. I guess I don't need to submit a new pull request, do I?
Thanks.

Comment (alalazo) [reply var/spack/repos/builtin/packages/z-checker/package.py]: No, that's absolutely fine to continue in this PR.

**Comment by alalazo** [comment]
Two minor issues to be solved, reported by `flake8`:
```
Dependencies found.
=======================================================
flake8: running flake8 code checks on spack.

Modified files:
  var/spack/repos/builtin/packages/sz/package.py
  var/spack/repos/builtin/packages/z-checker/package.py
=======================================================
var/spack/repos/builtin/packages/sz/package.py:135: [W293] blank line contains whitespace
var/spack/repos/builtin/packages/z-checker/package.py:34: [W293] blank line contains whitespace
```
and then I think it's ready to be merged.

File: var/spack/repos/builtin/packages/sz/package.py [hunk 2]
```diff
@@ -10,16 +10,10 @@ class Sz(CMakePackage):
     """Error-bounded Lossy Compressor for HPC Data"""
 
     homepage = "https://szcompressor.org"
-    url      = "https://github.com/szcompressor/SZ/archive/v2.1.10.tar.gz"
+    url      = "https://github.com/szcompressor/SZ/archive/v2.1.11.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/sz/package.py:13]: If you update this URL to `https://github.com/szcompressor/SZ/releases/download/v2.1.11/SZ-2.1.11.tar.gz` then the `url_for_version` shouldn't be needed, right?

File: var/spack/repos/builtin/packages/z-checker/package.py [hunk 3]
```diff
@@ -30,3 +31,9 @@ def configure_args(self):
         else:
             args += ['--disable-mpi']
         return args
+ 
+    def url_for_version(self, version):
+        """provide url to ensure that download counting via github releases
+        works accurately"""
+        url = "https://github.com/CODARcode/Z-checker/releases/download/{0}/Z-checker-{0}.tar.gz"
+        return url.format(version)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/z-checker/package.py:39]: Why is this needed? Shouldn't it already download from this URL?

Comment (robertu94) [reply var/spack/repos/builtin/packages/z-checker/package.py:39]: @disheng222 If I recall, if you use a url like https://github.com/szcompressor/SZ/releases/download/v2.1.11/SZ-2.1.11.tar.gz you get the release tracking information.  Is this correct?

**Comment by alalazo** [comment]
@disheng222 What is the status of this PR?

**Comment by robertu94** [comment]
@alalazo I believe it just needs to merged, but we should double check with @adamjstewart

**Comment by adamjstewart** [comment]
My only confusion is the URL. If the `url` attribute is set correctly, the `url_for_version` shouldn't be needed, right?

**Comment by robertu94** [comment]
@disheng222 and @adamjstewart ill look at this tonight and make sure that we have a correct patch
